src/main/java/httpserver/MethodWrapper.java

```diff
@@ -19,500 +21,147 @@
-        b.append("Too many available methods...\nOptional Methods:");
-        for (Method m : potentialMethods) {
-          b.append("\n\t");
-          b.append(m.toGenericString());
         }
-        throw new HTTPException(b.toString());
-      }
-      
-      method = potentialMethods.get(0);
-    }
-    catch(ClassNotFoundException | SecurityException e) {
-      throw new HTTPException("Could not add path.\n\tPath: " + path + "\n\tMethod: " + methodName, e);
     }
-  }
-
-
-  /**
-   * Invoke the method.
-   *
-   * @param callingClass  The class the method belongs to.
-   * @param path          The path that caused the method to be called. This is
-   *                      where variables come from.
-   *
-   * @throws HTTPException  If anything bad happened in invoking the underlying
-   *                        method. Probably shouldn't happen, because the
-   *                        issues would be found first when making the
-   *                        MethodWrapper, but there's a chance they could
-   *                        happen.
-   *
-   * @see java.lang.reflect.Method#invoke
-   */
-  public void invoke(Object callingClass, HTTPResponse resp, HTTPRequest req, String path) throws HTTPException {
-    List<Object> params = new ArrayList<>();
-    
-    try {
-      if (methodPath.equals("*")) {
-    	method.invoke(callingClass, resp, req);
-    	return;
-      }
-      
-      String[] methodPaths = methodPath.split("/");
-      
-      path = cleanPath(path);
-      
-      // Get the parameters
-      String[] paths;
-      if (path.length() == 0)
-    	paths = new String[0];
-      else
-    	paths = path.split("/");
-      
-      for (int i = 0; i < paths.length; i++) {
-        if (isDynamic(methodPaths[i])) {
-          if (!isArray(methodPaths[i])) {
-            Class<?> paramClass = Class.forName(LANG_PATH
-                    + getParamType(methodPaths[i]));
 
-            Constructor<?> paramConstructor
-                    = paramClass.getConstructor(String.class);
-
-            params.add(paramConstructor.newInstance(paths[i]));
-          } else { // we have an array
-            // Figure out the class type / constructor.
-            Class<?> paramClass =
-                    Class.forName(LANG_PATH + getParamType(methodPaths[i]));
-            Constructor<?> paramConstructor =
-                    paramClass.getConstructor(String.class);
-
-            // Create an arraylist so we can have variable sizes.
-            ArrayList<Object> list = new ArrayList<>();
-
-            // Place the values in our list.
-            for(int j=i; j<paths.length; j++) {
-              list.add(paramConstructor.newInstance(paths[j]));
+    /**
+     * Call the route.
+     *
+     * @param path          The path that caused the method to be called. This is
+     *                      where variables come from.
+     */
+    public void invoke(HTTPRequest request, HTTPResponse response, List<String> calledPath) {

```

Comment:
I'm not sure I see what you're saying.

Do you think the `HTTPHandler#handle` method should do something like:

``` java
Map<String, String> params = method.extractParameters(calledPath);
request.mergeParams(params);

List<String> varargs = method.extractVarargs(calledPath);
request.mergeVarargs(varargs);

method.invoke(); // or maybe rename to method.handle() / method.call() ?
```

?